### PR TITLE
mcs: fix the scatter error

### LINKED ISSUE #7234: The error of region not found is not properly convert to response
### What did you do?

Run br/lightning tests 

### What did you expect to see?

Test passed

### What did you see instead?

```
[2023/10/19 07:33:38.050 +00:00] [WARN] [localhelper.go:487] ["wait for scatter region encountered non-retryable error"] [region="{ID=253,startKey=7480000000000000FF6B00000000000000F8,endKey=7480000000000000FF6D00000000000000F8,epoch=\"conf_ver:5 version:63 \",peers=\"id:254 store_id:1 ,id:255 store_id:12 ,id:256 store_id:14 \"}"] [error="failed to get region operator, error type: UNKNOWN, error message: Not Found"] [errorVerbose="failed to get region operator, error type: UNKNOWN, error message: Not Found\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).checkRegionScatteredOrReScatter\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/localhelper.go:526\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).waitForScatterRegions.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/localhelper.go:480\ngithub.com/pingcap/tidb/br/pkg/utils.WithRetry.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/utils/retry.go:57\ngithub.com/pingcap/tidb/br/pkg/utils.WithRetryV2[...]\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/utils/retry.go:75\ngithub.com/pingcap/tidb/br/pkg/utils.WithRetry\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/utils/retry.go:56\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).waitForScatterRegions\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/localhelper.go:477\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).SplitAndScatterRegionByRanges\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/localhelper.go:357\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).SplitAndScatterRegionInBatches\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/localhelper.go:123\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).prepareAndSendJob\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/local.go:1136\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).doImport\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/local.go:1683\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).ImportEngine\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/local.go:1558\ngithub.com/pingcap/tidb/br/pkg/lightning/backend.(*ClosedEngine).Import\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/backend.go:365\ngithub.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importKV\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:1242\ngithub.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importEngine\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:891\ngithub.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importEngines.func3\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:504\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1650"]
[2023/10/19 07:33:38.051 +00:00] [INFO] [localhelper.go:362] ["waiting for scattering regions timeout"] [scatterCount=0] [regions=1] [take=619.326<C2><B5>s] [error="failed to get region operator, error type: UNKNOWN, error message: Not Found"] [errorVerbose="failed to get region operator, error type: UNKNOWN, error message: Not Found\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).checkRegionScatteredOrReScatter\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/localhelper.go:526\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).waitForScatterRegions.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/localhelper.go:480\ngithub.com/pingcap/tidb/br/pkg/utils.WithRetry.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/utils/retry.go:57\ngithub.com/pingcap/tidb/br/pkg/utils.WithRetryV2[...]\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/utils/retry.go:75\ngithub.com/pingcap/tidb/br/pkg/utils.WithRetry\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/utils/retry.go:56\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).waitForScatterRegions\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/localhelper.go:477\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).SplitAndScatterRegionByRanges\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/localhelper.go:357\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).SplitAndScatterRegionInBatches\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/localhelper.go:123\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).prepareAndSendJob\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/local.go:1136\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).doImport\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/local.go:1683\ngithub.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Backend).ImportEngine\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/local.go:1558\ngithub.com/pingcap/tidb/br/pkg/lightning/backend.(*ClosedEngine).Import\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/backend.go:365\ngithub.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importKV\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:1242\ngithub.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importEngine\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:891\ngithub.com/pingcap/tidb/br/pkg/lightning/importer.(*TableImporter).importEngines.func3\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/importer/table_import.go:504\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1650"]
```

### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #7234.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Integration test

https://tcms.pingcap.net/dashboard/executions/plan/3450271

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]

File: pkg/schedule/scatter/region_scatterer.go [hunk 1]
```diff
@@ -192,6 +192,9 @@ func (r *RegionScatterer) ScatterRegionsByID(regionsID []uint64, group string, r
 		region := r.cluster.GetRegion(id)
 		if region == nil {
 			scatterSkipNoRegionCounter.Inc()
+			if len(regionsID) == 1 {
+				return 0, nil, errors.New("region not found")
+			}
```
Comment (lhy1024) [inline pkg/schedule/scatter/region_scatterer.go:197]: how about moving it to L188
```
                        if len(regionsID) == 1 {
                                if r.cluster.GetRegion(regionsID[0])==nil{
                                           scatterSkipNoRegionCounter.Inc()
				           return 0, nil, errors.New("region not found")
                                }
			}
```

File: server/grpc_service.go [hunk 2]
```diff
@@ -1802,7 +1813,14 @@ func (s *GrpcServer) ScatterRegion(ctx context.Context, request *pdpb.ScatterReg
...
 		if forwardCli != nil {
-			regionsID := request.GetRegionsId()
+			var regionsID []uint64
+			// nolint
+			if request.GetRegionId() != 0 {
+				// nolint
+				regionsID = []uint64{request.GetRegionId()}
+			} else {
+				regionsID = request.GetRegionsId()
+			}
```
Comment (lhy1024) [inline server/grpc_service.go:1823]: why need nolint?

File: server/grpc_service.go [hunk 3]
```diff
@@ -2134,6 +2152,15 @@ func (s *GrpcServer) invalidValue(msg string) *pdpb.ResponseHeader {
 func (s *GrpcServer) convertHeader(header *schedulingpb.ResponseHeader) *pdpb.ResponseHeader {
 	switch header.GetError().GetType() {
 	case schedulingpb.ErrorType_UNKNOWN:
+		if strings.Contains(header.GetError().GetMessage(), "region not found") {
```
Comment (lhy1024) [inline server/grpc_service.go:2155]: how about using const value for "region not found"

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: server/grpc_service.go (hunk 2 above)
Comment (rleungx) [reply server/grpc_service.go:1823]: GetRegionId is deprecated

File: pkg/schedule/scatter/region_scatterer.go [hunk 4]
```diff
@@ -184,17 +186,18 @@ func (r *RegionScatterer) ScatterRegionsByRange(startKey, endKey []byte, group s
 func (r *RegionScatterer) ScatterRegionsByID(regionsID []uint64, group string, retryLimit int, skipStoreLimit bool) (int, map[uint64]error, error) {
 	if len(regionsID) < 1 {
 		scatterSkipEmptyRegionCounter.Inc()
-		return 0, nil, errors.New("empty region")
+		return 0, nil, errEmptyRegion
+	}
+	if len(regionsID) == 1 {
```
Comment (lhy1024) [inline pkg/schedule/scatter/region_scatterer.go:191]: It's different from previous, do we need to check whether it is nil?

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**